src/main/java/de/dhbw/karlsruhe/models/ProductionSet.java

```diff
@@ -0,0 +1,82 @@
+package de.dhbw.karlsruhe.models;
+
+import java.util.*;
+
+public class ProductionSet {
+    Set<GrammarRule> grammarRules = new HashSet<>();
+
+    public ProductionSet(GrammarRule startRule){
+        grammarRules.add(startRule);
+    }
+
+    public ProductionSet(List<GrammarRule> grList){
+        grammarRules.addAll(grList);
+    }
+
+    public void addProduction(GrammarRule gr){
+        if (isOnRightSide(gr.leftSide()) || isOnLeftSide(gr.leftSide())){
```

Comment:
Was wird hier genau geprüft?

Reply:
Hier wird geprüft, ob in einer bestehenden Produktion das linke Nichtterminal der neuen Produktion auf der rechten oder auf der linken Seite bereits vorkommt. Damit wird geprüft, ob die eingegebene Produktion von den bestehenden Produktionen erreicht werden kann. Sollte man das im Code erläutern?

Reply:
Wenn es links schon irgendwo vorkommt muss es ja nicht zwangsläufig erreichbar sein oder? Es ist ja nur erreichbar, wenn es rechts vorkommt.

Reply:
Da nur Produktionen im Set sind die bereits erreichbar sind gilt das prüfen der linken Seite auch. Das muss geprüft werden, falls das Nichtterminal der Produtkion das Startnichtterminal der Grammatik ist und somit vielleicht auf keiner rechten Seite vorkommt.